### PR TITLE
feat(ui): Migrate project summary settings to EditablePanel for parity with rest of UI

### PR DESCRIPTION
**Pull request by rbreeze**
Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [x] Optional. My organization is added to USERS.md.
* [x] I've signed the CLA and my build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

### CODE REVIEW

File: ui/src/app/shared/models.ts [hunk 1]
```diff
@@ -721,3 +721,59 @@ export interface GnuPGPublicKey {
 }
 
 export interface GnuPGPublicKeyList extends ItemsList<GnuPGPublicKey> {}
+
+// https://kubernetes.io/docs/reference/kubectl/overview/#resource-types
+
+export enum ResourceKinds {
+    ANY = '*',
```
Comment (alexmt) [inline ui/src/app/shared/models.ts:728]: Looks like we are using only keys of that map, but not values. The `ANY` is now supposed to be suggested to the user. Should we just store keys are an array and replace `ANY` with `*` ?

File: ui/src/app/settings/components/project-details/project-details.tsx [hunk 2]
```diff
@@ -473,225 +430,505 @@ export class ProjectDetails extends React.Component<RouteComponentProps<{name: s
...
+                                            <div className='columns small-4'>
+                                                <FormField
+                                                    formApi={formApi}
+                                                    field={`spec.clusterResourceBlacklist[${i}].kind`}
+                                                    component={AutocompleteField}
+                                                    componentProps={{items: Object.keys(ResourceKinds)}}
+                                                />
+                                            </div>
+                                            <div className='columns small-8'>
+                                                <FormField formApi={formApi} field={`spec.clusterResourceBlacklist[${i}].group`} component={AutocompleteField} />
```
Comment (alexmt) [inline ui/src/app/settings/components/project-details/project-details.tsx:656]: Same as K8S Kind we can suggest k8s groups.

File: ui/src/app/settings/components/project-details/project-details.tsx [hunk 3]
```diff
@@ -473,225 +430,505 @@ export class ProjectDetails extends React.Component<RouteComponentProps<{name: s
...
-                                <div className='row'>
-                                    <div className='columns small-3'>{dst.server}</div>
-                                    <div className='columns small-6'>{dst.namespace}</div>
+                <EditablePanel
+                    save={item => this.saveProject(item)}
+                    values={proj}
+                    title='SOURCE REPOSITORIES'
+                    view={
+                        <React.Fragment>
+                            {(proj.spec.sourceRepos || []).map((repo, i) => (
```
Comment (alexmt) [inline ui/src/app/settings/components/project-details/project-details.tsx:481]: Please add an empty state message here and in other panels

**Review by alexmt — CHANGES_REQUESTED** [review]
Thanks a lot for working on it @rbreeze !

Added few comments. Also, the orphaned resources panel seems to need a little more polishing. Worked on small PR that implements some ideas: https://github.com/rbreeze/argo-cd/pull/1

* add icons that visualize enabled/disabled state
* add help icons with the additional description in tooltip
* removed redundant 'Orphaned' words from labels within 'Orphaned Resources' panel
* make sure that positions of edit and view controller are consistent 

image


image

File: ui/src/app/shared/models.ts (hunk 1 above)
Comment (rbreeze) [reply ui/src/app/shared/models.ts:728]: Sounds good. The reason I created this as an enum was for type safety in a ResourceKindSelector component - I'll convert to array for now and create the type in refactor PR

File: ui/src/app/shared/models.ts [hunk 4]
```diff
@@ -721,3 +721,57 @@ export interface GnuPGPublicKey {
 }
 
 export interface GnuPGPublicKeyList extends ItemsList<GnuPGPublicKey> {}
+
+// https://kubernetes.io/docs/reference/kubectl/overview/#resource-types
+
+export const ResourceKinds = [
```
Comment (alexmt) [inline ui/src/app/shared/models.ts:727]: ResourcesKinds should include capitalized singular kinds like `Pod`, `Deployment` etc.

File: ui/src/app/settings/components/project-details/project-details.tsx [hunk 5]
```diff
@@ -473,225 +435,560 @@ export class ProjectDetails extends React.Component<RouteComponentProps<{name: s
...
+                                      <div className='row white-box__details-row' key={i}>
+                                          <div className='columns small-4'>{resource.kind}</div>
+                                          <div className='columns small-8'>{resource.group}</div>
+                                      </div>
+                                  ))
+                                : emptyMessage('cluster resource allow list')}
+                        </React.Fragment>
+                    }
+                    edit={formApi => (
+                        <DataLoader load={() => services.clusters.list()}>
```
Comment (alexmt) [inline ui/src/app/settings/components/project-details/project-details.tsx:596]: Looks like `clusters` is not used here. Please remove `<DataLoader load={() => services.clusters.list()}>`

File: ui/src/app/settings/components/project-details/project-details.tsx [hunk 6]
```diff
@@ -473,225 +435,560 @@ export class ProjectDetails extends React.Component<RouteComponentProps<{name: s
...
+                                          <div className='columns small-4'>{resource.kind}</div>
+                                          <div className='columns small-8'>{resource.group}</div>
+                                      </div>
+                                  ))
+                                : emptyMessage('cluster resource deny list')}
+                        </React.Fragment>
+                    }
+                    edit={formApi => (
+                        <DataLoader load={() => services.clusters.list()}>
+                            {clusters => (
```
Comment (alexmt) [inline ui/src/app/settings/components/project-details/project-details.tsx:655]: Looks like `clusters` is not used here. Please remove `<DataLoader load={() => services.clusters.list()}>`

File: ui/src/app/settings/components/project-details/project-details.tsx [hunk 7]
```diff
@@ -473,225 +435,560 @@ export class ProjectDetails extends React.Component<RouteComponentProps<{name: s
...
-                )) || (
-                    <div className='white-box'>
-                        <p>No cluster-scoped resources are not permitted to deploy</p>
-                    </div>
-                )}
+                <EditablePanel
+                    save={item => this.saveProject(item)}
+                    values={proj}
+                    title={<React.Fragment>DESTINATIONS {helpTip('Cluster and namespaces where applications are permitted to be deployed to')}</React.Fragment>}
+                    view={
```
Comment (alexmt) [inline ui/src/app/settings/components/project-details/project-details.tsx:527]: Add header please so user can understand which means of each column:

```htmp
                    <React.Fragment>
                         <div className='row white-box__details-row'>
                                <div className='columns small-4'>Server</div>
                                <div className='columns small-8'>Namespace</div>
                            </div>
```

image

File: ui/src/app/settings/components/project-details/project-details.tsx [hunk 8]
```diff
@@ -473,225 +435,560 @@ export class ProjectDetails extends React.Component<RouteComponentProps<{name: s
...
+                                      </div>
+                                  ))
+                                : emptyMessage('destinations')}
+                        </React.Fragment>
+                    }
+                    edit={formApi => (
+                        <DataLoader load={() => services.clusters.list()}>
+                            {clusters => (
+                                <React.Fragment>
+                                    {(formApi.values.spec.destinations || []).map((_: Project, i: number) => (
```
Comment (alexmt) [inline ui/src/app/settings/components/project-details/project-details.tsx:543]: Add header please so user can understand which means of each column.

**Review by alexmt — CHANGES_REQUESTED** [review]
Last couple comments. LGTM after addressing them.

**Review by alexmt — APPROVED** [review]
LGTM. Thank you @rbreeze

File: ui/src/app/settings/components/project-details/project-details.tsx [hunk 9]
```diff
@@ -473,225 +435,627 @@ export class ProjectDetails extends React.Component<RouteComponentProps<{name: s
...
+                                ADD RESOURCE
+                            </button>
+                        </React.Fragment>
+                    )}
+                    items={[]}
+                />
+                <EditablePanel
+                    save={item => this.saveProject(item)}
+                    values={proj}
+                    title={<React.Fragment>CLUSTER RESOURCE DENY LIST {helpTip('Cluster-scoped K8s API Groups and Kinds which are not permitted to be deployed')}</React.Fragment>}
```
Comment (jsoref) [inline ui/src/app/settings/components/project-details/project-details.tsx:665]: It feels like this change from `blacklist`/`whitelist` to `deny list`/`allow list` should have been called out in a commit message / release note...

**Pull request merged**